src/main/java/br/com/caelum/viagens/voos/validator/RotaRepetidaValidator.java

```diff
@@ -0,0 +1,40 @@
+package br.com.caelum.viagens.voos.validator;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import br.com.caelum.viagens.administrativo.model.Rota;
+import br.com.caelum.viagens.administrativo.repository.RotaRepository;
+import br.com.caelum.viagens.voos.controller.dto.input.NewVooInputDto;
+
+public class RotaRepetidaValidator implements Validator{
+	
+	private RotaRepository rotaRepository;
+	
+	public RotaRepetidaValidator(RotaRepository rotaRepository) {
+		this.rotaRepository = rotaRepository;
+	}
+			
+	@Override
+	public boolean supports(Class<?> clazz) {
+		return NewVooInputDto.class.isAssignableFrom(clazz);
+	}
+
+	@Override
+	public void validate(Object target, Errors errors) {
+		NewVooInputDto newVooDto = (NewVooInputDto) target;
+		
+		Set<Optional<Rota>> conjRotas = newVooDto.getRotas().stream()
```

Comment:
To aqui pensando...pelo que lembro você já recebe as rotas como um Set lá no form. Pq teria repetida? A própria coleção já ia comer isso... Se você recebesse como uma lista, teria a possibilidade... mas aí também daria para você fazer a lógica por dentro da lista apenas com os ids das rotas. Que acha? Faz sentido?

Reply:
"pq teria repetida?" eu não escrevi o equals no dto propositalmente já que eu quero mostrar a mensagem para o cliente se ele passar uma rota repetida - quero deixar claro que isso não é válido. eu modifiquei para ir apenas uma vez ao banco, mas acrescentei um if perguntando se não existe mais nenhum erro já que é preciso verificar antes se a rota existe para ele funcionar como o esperado. isso também faz sentido?